File: #59_Spiral_Matrix_II/Solution.py
class Solution:
	# @param {integer} n
	# @return {integer[][]}
	def generateMatrix(self, n):
		if n == 0:
			return []
		matrix = [[0 for j in range(n)] for i in range(n)]
		startRow = 0
		endRow = n - 1
		startCol = 0
		endCol = n - 1
		num = 1
		while startRow <= endRow:
			for j in range(startCol, endCol + 1):
				matrix[startRow][j] = num
				num += 1
			startRow += 1
			for i in range(startRow, endRow + 1):
				matrix[i][endCol] = num
				num += 1
			endCol -= 1
			for j in range(endCol, startCol - 1, -1):
				matrix[endRow][j] = num
				num += 1
			endRow -= 1
			for i in range(endRow, startRow - 1, -1):
				matrix[i][startCol] = num
				num += 1
			startCol += 1
		return matrix

sol = Solution()
print sol.generateMatrix(1)

File: #59_Spiral_Matrix_II/Solution.java
public class Solution {
    public int[][] generateMatrix(int n) {
        int[][] result = new int[n][n];
    	if (n == 0) return result;
        if (n == 1) {
        	result[0][0] = 1;
        	return result;
        }
        int max = n * n;
        int index = 1;
        int level = n;
    	int startRow = 0; int startCol = 0;
        while (index <= max) {
        	int i = startRow; int j = startCol;
        	while (j < level) {
        		result[i][j] = index++;
        		j++;
        	}
        	i++; j--;
        	while (i < level) {
        		result[i][j] = index++;
        		i++;
        	}
        	i--; j--;
        	while (j >= startCol) {
        		result[i][j] = index++;
        		j--;
        	}
        	j++; i--;
        	while (i > startRow) {
        		result[i][j] = index++;
        		i--;
        	}
        	startRow++; startCol++; level -= 1;
        }
        return result;
    }

    public static void main(String[] args) {
    	Solution sol = new Solution();
    	int[][] result = sol.generateMatrix(0);
    	for (int i = 0; i < result.length; i++) {
    		for (int j = 0; j < result[0].length; j++) {
    			System.out.print(result[i][j] + " ");
    		}
    		System.out.println();
    	}
    }
}